### PR TITLE
Overload user mgmt API

### PR DESCRIPTION
**Pull request by asafshen**
## Related Issues

Fixes https://github.com/descope/etc/issues/5316

## Description
overlaod user mgmt API in a a manner that is not breaking changes

also TS know to autocomplete well
image

### CODE REVIEW

File: lib/management/user.ts [hunk 1]
```diff
@@ -52,41 +54,61 @@ const withUser = (sdk: CoreSdk, managementKey?: string) => ({
...
+    picture?: string,
+    verifiedEmail?: boolean,
+    verifiedPhone?: boolean,
+    givenName?: string,
+    middleName?: string,
+    familyName?: string,
+    additionalLoginIds?: string[],
+  ): Promise<SdkResponse<UserResponse>> {
+    const body =
+      typeof emailOrOptions === 'string'
```
Comment (asafshen) [inline lib/management/user.ts:76]: I wonder if we should add some deprecation message here, let me know what you think

Comment (dorsha) [reply lib/management/user.ts:76]: Not sure if deprecation, but maybe explain the rational here.

**Review by dorsha — COMMENTED** [review]
Very nice!!!

File: lib/management/user.ts (hunk 1 above)
Comment (asafshen) [reply lib/management/user.ts:79]: added 👍

**Review by dorsha — APPROVED** [review]

**Pull request merged**